imap-proto/src/parser/core.rs

```diff
@@ -1,5 +1,4 @@
-use nom::{self, IResult};
-use std::str;
+use nom::{character::streaming::digit1, IResult};
```

Comment:
It's not clear why `digit1` gets imported here, but `std::str` has to go (I do think it makes sense to import `digit1`, but doing both seems somewhat inconsistent). I'd prefer to keep the `std::str` import, unless you have a good argument why we shouldn't use it. (Potentially we should import `std::str::from_utf8()` directly if that's the only thing being used from `std::str`.)